### ISSUE TITLE
XML messages from the where command are not well-formed causing server disconnect

### ISSUE THREAD

**Issue by GoogleCodeExporter** [baluragala/gtalksms#81]
```
User reported this error which I could reproduce. Issuing the "where" command 
with OSM enabled disconnects gtalksms. Disconnect is issued by the server 
because of the not well-formed xml messages.


```

Original issue reported on code.google.com by `fschm...@gmail.com` on 7 Feb 2011 at 12:03

**Comment by GoogleCodeExporter** [comment]
Original comment by `fschm...@gmail.com` on 7 Feb 2011 at 12:03
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
For information, it works fine on my phone (with pidgin)
```

Original comment by `Florent....@gmail.com` on 7 Feb 2011 at 12:06

**Comment by GoogleCodeExporter** [comment]
```
It's not a xmpp client problem. It depends on the xmpp server you use for the 
connection, to be more precise, if the server checks if the xml-messages are 
well-formed.
```

Original comment by `fschm...@gmail.com` on 7 Feb 2011 at 12:08

**Comment by GoogleCodeExporter** [comment]
```
[org.jivesoftware.openfire.nio.ConnectionHandler.messageReceived(ConnectionHandl
er.java:135)] Closing connection due to error while processing message: 
<message id="B232f-12" to="flo@freakempire.de" 
type="chat"><body>http://maps.google.com/maps?q=37.422005,-122.084095
http://www.openstreetmap.org/?mlat=37.422005&mlon=-122.084095&zoom=14&layers=M
Genauigkeit: 0,000000m<C2><A0>
H<C3><B6>he: 0,000000<C2><A0>
Geschwindigkeit: 0,000000m/s<C2><A0>
Quelle: gps</body><html 
xmlns="http://jabber.org/protocol/xhtml-im"><body><font>http://maps.google.com/m
aps?q=37.422005,-122.084095<br/>
http://www.openstreetmap.org/?mlat=37.422005&mlon=-122.084095&zoom=14&layers=M<b
r/>
Genauigkeit: 0,000000m <br/>
Höhe: 0,000000 <br/>
Geschwindigkeit: 0,000000m/s <br/>
Quelle: gps</font></body></html></message>
org.xmlpull.v1.XmlPullParserException: entity reference name can not contain 
character =' (position: TEXT seen 
...<br/>\nhttp://www.openstreetmap.org/?mlat=37.422005&mlon=... @7:51) 

```

Original comment by `fschm...@gmail.com` on 7 Feb 2011 at 12:33

**Comment by GoogleCodeExporter** [comment]
```
Florent: Patch is in my clone: 
http://code.google.com/r/fschmaus-gtalksms/source/detail?r=8d91c1c28e2681a5cd4cd
b85966ab5255ee5109d#
Please give feedback and if you have an questions just ask. :-)

```

Original comment by `fschm...@gmail.com` on 7 Feb 2011 at 4:50
- Changed state: **WaitingForRelease**

**Comment by GoogleCodeExporter** [comment]
```
That seems good for me.
```

Original comment by `Florent....@gmail.com` on 7 Feb 2011 at 11:21

**Comment by GoogleCodeExporter** [comment]
```
Fixed in v1.9.1
```

Original comment by `Florent....@gmail.com` on 13 Feb 2011 at 12:48
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
Original comment by `fschm...@gmail.com` on 25 Aug 2011 at 7:51
- Changed state: **Deployed**
